pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -148,6 +149,32 @@ func validateContainer(container corev1.Container) *apis.FieldError {
 	return errs
 }
 
+func validateContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
+	// user can set container port which names "user-port" to define application's port.
+	// Queue-proxy will use it to send requests to application
+	// if user didn't set any port, it will set default port user-port=8080.
+	if len(ports) > 1 {
+		return &apis.FieldError{
+			Message: "More than one container ports is set",
+			Paths:   []string{apis.CurrentField},
+			Details: "Only a single unnamed port is allowed",
+		}
+	}
+
+	// the port is named "user-port" on the deployment, but a user cannot set a name on the port
+	// in Configuration. The name field is reserved for content-negotiation. Since content-negotiation
+	// is not yet implemented we do not allow any name.
+	// https://github.com/knative/serving/blob/master/docs/runtime-contract.md#inbound-network-connectivity
```

Comment:
can we start allowing names here, but limited to the set of names allowed in spec, to unblock @tanzeeb?

Reply:
That would be great! Specifically, `h2c` and `http1`.

Reply:
Yes I can do that. I was not sure when this was going to be implemented and didn't want behavior to change for users that put a name so went with the safest option. Since this is following so close I will open up 'h2c' and 'http1'.